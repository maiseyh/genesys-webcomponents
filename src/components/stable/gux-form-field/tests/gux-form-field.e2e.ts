import { newE2EPage } from '@stencil/core/testing';

describe('gux-form-field', () => {
  it('renders', async () => {
    const html = `
      <gux-form-field lang="en">
        <input slot="input" type="text">
        <label slot="label">Text</label>
      </gux-form-field>
    `;
    const page = await newE2EPage({ html });

    const element = await page.find('gux-form-field');
    expect(element).toHaveClass('hydrated');
  });

  describe('#render', () => {
    describe('input tag', () => {
      [
        { type: 'checkbox', value: 'true' },
        { type: 'color', value: '#000000' },
        { type: 'email', value: 'test@example.com' },
        { type: 'number', value: '2' },
        { type: 'password', value: 'text' },
        { type: 'radio', value: 'true' },
        { type: 'range', value: '5' },
        { type: 'search', value: 'text' },
        { type: 'text', value: 'text' },
        { type: 'unknown', value: 'text' }
      ].forEach(({ type, value }) => {
        it(`should render component type "${type}"`, async () => {
          const html = `
            <gux-form-field lang="en">
              <input
                slot="input"
                type="${type}"
                id="test"
                value="${value}"
              />
              <label slot="label" for="test">Test</label>
            </gux-form-field>
          `;
          const page = await newE2EPage({ html });
          const element = await page.find('gux-form-field');

          expect(element.innerHTML).toMatchSnapshot();
        });
      });
    });

    describe('seect tag', () => {
      it(`should render component type "select"`, async () => {
        const html = `
          <gux-form-field lang="en">
            <select slot="input" name="select" type="select">
              <option value="option1">Option 1</option>
              <option value="option1">Option 2</option>
              <option value="option1">Option 3</option>
            </select>
            <label slot="label">Select</label>
          </gux-form-field>
        `;
        const page = await newE2EPage({ html });
        const element = await page.find('gux-form-field');

        expect(element.innerHTML).toMatchSnapshot();
      });
    });
  });
});
