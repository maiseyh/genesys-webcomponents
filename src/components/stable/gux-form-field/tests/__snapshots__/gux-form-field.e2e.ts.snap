// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-form-field #render input tag should render component type "checkbox" 1`] = `"<!----> <gux-input-checkbox class=\\"hydrated\\"><!----><div class=\\"gux-input-checkbox-container\\"><input slot=\\"input\\" type=\\"checkbox\\" id=\\"test\\" value=\\"true\\"><label slot=\\"label\\" for=\\"test\\">Test</label></div></gux-input-checkbox>"`;

exports[`gux-form-field #render input tag should render component type "color" 1`] = `"<!----> <div class=\\"gux-label-and-input-and-error-container\\"><div class=\\"gux-label-and-input-container gux-beside\\"><label slot=\\"label\\" for=\\"test\\">Test</label><gux-input-color class=\\"hydrated\\"><!----><section><button type=\\"button\\" class=\\"gux-input-color-main-element\\"><div class=\\"gux-input-color-selected-color\\" style=\\"background: rgb(0, 0, 0);\\"></div><div class=\\"gux-input-color-color-name\\">#000000</div><gux-icon icon-name=\\"chevron-small-down\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M11 6L8 9 5 6 4 7l4 4 4-4z\\"></path></svg></div></gux-icon></button><gux-color-select class=\\"gux-input-color-color-select hydrated\\"><!----><div hidden=\\"\\"><input slot=\\"input\\" type=\\"color\\" id=\\"test\\" value=\\"#000000\\"></div><div><div class=\\"gux-input-color-matrix\\"><gux-input-color-option class=\\"hydrated\\"><button type=\\"button\\" value=\\"#203B73\\" title=\\"#203B73\\" style=\\"background-color: rgb(32, 59, 115);\\"></button></gux-input-color-option><gux-input-color-option class=\\"hydrated\\"><button type=\\"button\\" value=\\"#75A8FF\\" title=\\"#75A8FF\\" style=\\"background-color: rgb(117, 168, 255);\\"></button></gux-input-color-option><gux-input-color-option class=\\"hydrated\\"><button type=\\"button\\" value=\\"#8452CF\\" title=\\"#8452CF\\" style=\\"background-color: rgb(132, 82, 207);\\"></button></gux-input-color-option><gux-input-color-option class=\\"hydrated\\"><button type=\\"button\\" value=\\"#1DA8B3\\" title=\\"#1DA8B3\\" style=\\"background-color: rgb(29, 168, 179);\\"></button></gux-input-color-option><gux-input-color-option class=\\"hydrated\\"><button type=\\"button\\" value=\\"#B5B5EB\\" title=\\"#B5B5EB\\" style=\\"background-color: rgb(181, 181, 235);\\"></button></gux-input-color-option><gux-input-color-option class=\\"hydrated\\"><button type=\\"button\\" value=\\"#CC3EBE\\" title=\\"#CC3EBE\\" style=\\"background-color: rgb(204, 62, 190);\\"></button></gux-input-color-option><gux-input-color-option class=\\"hydrated\\"><button type=\\"button\\" value=\\"#5E6782\\" title=\\"#5E6782\\" style=\\"background-color: rgb(94, 103, 130);\\"></button></gux-input-color-option><gux-input-color-option class=\\"hydrated\\"><button type=\\"button\\" value=\\"#FF8FDD\\" title=\\"#FF8FDD\\" style=\\"background-color: rgb(255, 143, 221);\\"></button></gux-input-color-option><gux-input-color-option class=\\"hydrated\\"><button type=\\"button\\" value=\\"#868C1E\\" title=\\"#868C1E\\" style=\\"background-color: rgb(134, 140, 30);\\"></button></gux-input-color-option><gux-input-color-option class=\\"hydrated\\"><button type=\\"button\\" value=\\"#DDD933\\" title=\\"#DDD933\\" style=\\"background-color: rgb(221, 217, 51);\\"></button></gux-input-color-option></div></div></gux-color-select></section></gux-input-color></div><div class=\\"gux-error\\"></div></div>"`;

exports[`gux-form-field #render input tag should render component type "email" 1`] = `"<!----> <div class=\\"gux-label-and-input-and-error-container\\"><div class=\\"gux-label-and-input-container gux-beside\\"><label slot=\\"label\\" for=\\"test\\">Test</label><gux-input-text-like slot=\\"input\\" class=\\"hydrated\\"><!----><div class=\\"gux-input-container\\"><input slot=\\"input\\" type=\\"email\\" id=\\"test\\" value=\\"test@example.com\\"></div></gux-input-text-like></div><div class=\\"gux-error\\"></div></div>"`;

exports[`gux-form-field #render input tag should render component type "number" 1`] = `"<!----> <div class=\\"gux-label-and-input-and-error-container\\"><div class=\\"gux-label-and-input-container gux-beside\\"><label slot=\\"label\\" for=\\"test\\">Test</label><gux-input-number slot=\\"input\\" class=\\"hydrated\\"><!----><div class=\\"gux-input-number-container\\"><div class=\\"gux-input-container\\"><input slot=\\"input\\" type=\\"number\\" id=\\"test\\" value=\\"2\\"></div><div class=\\"gux-step-buttons-container\\"><button class=\\"gux-step-button\\" type=\\"button\\" title=\\"Increment\\"><gux-icon icon-name=\\"chevron-small-up\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M8 5L4 9l1 1 3-3 3 3 1-1z\\"></path></svg></div></gux-icon></button><button class=\\"gux-step-button\\" type=\\"button\\" title=\\"Decrement\\"><gux-icon icon-name=\\"chevron-small-down\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M11 6L8 9 5 6 4 7l4 4 4-4z\\"></path></svg></div></gux-icon></button></div></div></gux-input-number></div><div class=\\"gux-error\\"></div></div>"`;

exports[`gux-form-field #render input tag should render component type "password" 1`] = `"<!----> <div class=\\"gux-label-and-input-and-error-container\\"><div class=\\"gux-label-and-input-container gux-beside\\"><label slot=\\"label\\" for=\\"test\\">Test</label><gux-input-text-like slot=\\"input\\" class=\\"hydrated\\"><!----><div class=\\"gux-input-container\\"><input slot=\\"input\\" type=\\"password\\" id=\\"test\\" value=\\"text\\"></div></gux-input-text-like></div><div class=\\"gux-error\\"></div></div>"`;

exports[`gux-form-field #render input tag should render component type "radio" 1`] = `"<!----> <gux-input-radio class=\\"hydrated\\"><!----><div class=\\"gux-input-radio-container\\"><input slot=\\"input\\" type=\\"radio\\" id=\\"test\\" value=\\"true\\"><label slot=\\"label\\" for=\\"test\\">Test</label></div></gux-input-radio>"`;

exports[`gux-form-field #render input tag should render component type "range" 1`] = `"<!----> <div class=\\"guxlabel-and-input-and-error-container\\"><div class=\\"gux-label-and-input-container gux-beside\\"><label slot=\\"label\\" for=\\"test\\">Test</label><gux-input-range class=\\"hydrated\\"><!----><div class=\\"gux-container\\"><div class=\\"gux-range\\"><div class=\\"gux-track\\"><div class=\\"gux-progress\\" style=\\"width: 5%;\\"></div></div><input slot=\\"input\\" type=\\"range\\" id=\\"test\\" value=\\"5\\"><div class=\\"gux-range-tooltip-container\\"><div class=\\"gux-range-tooltip\\" style=\\"left: 4.92028%;\\">5</div></div></div><div class=\\"gux-display gux-hidden\\">5</div></div></gux-input-range></div><div class=\\"gux-error\\"></div></div>"`;

exports[`gux-form-field #render input tag should render component type "search" 1`] = `"<!----> <div class=\\"gux-label-and-input-and-error-container\\"><div class=\\"gux-label-and-input-container gux-beside\\"><label slot=\\"label\\" for=\\"test\\">Test</label><gux-input-text-like slot=\\"input\\" class=\\"hydrated\\"><!----><div class=\\"gux-input-container\\"><input slot=\\"input\\" type=\\"search\\" id=\\"test\\" value=\\"text\\"></div></gux-input-text-like></div><div class=\\"gux-error\\"></div></div>"`;

exports[`gux-form-field #render input tag should render component type "text" 1`] = `"<!----> <div class=\\"gux-label-and-input-and-error-container\\"><div class=\\"gux-label-and-input-container gux-beside\\"><label slot=\\"label\\" for=\\"test\\">Test</label><gux-input-text-like slot=\\"input\\" class=\\"hydrated\\"><!----><div class=\\"gux-input-container\\"><input slot=\\"input\\" type=\\"text\\" id=\\"test\\" value=\\"text\\"></div></gux-input-text-like></div><div class=\\"gux-error\\"></div></div>"`;

exports[`gux-form-field #render input tag should render component type "unknown" 1`] = `"<!----> <div><label slot=\\"label\\" for=\\"test\\">Test</label><input slot=\\"input\\" type=\\"unknown\\" id=\\"test\\" value=\\"text\\"></div>"`;

exports[`gux-form-field #render seect tag should render component type "select" 1`] = `"<!----> <div class=\\"gux-label-and-input-and-error-container\\"><div class=\\"gux-label-and-input-container gux-beside\\"><label slot=\\"label\\">Select</label><gux-input-select slot=\\"input\\" class=\\"hydrated\\"><!----><div id=\\"gux-input-select-p94cs922xb\\"><div class=\\"gux-slot\\"><select slot=\\"input\\" name=\\"select\\" type=\\"select\\"> <option value=\\"option1\\">Option 1</option> <option value=\\"option1\\">Option 2</option> <option value=\\"option1\\">Option 3</option> </select></div><button type=\\"button\\" class=\\"gux-field-button\\" aria-haspopup=\\"listbox\\"><div class=\\"gux-selected-option\\">Option 1</div><gux-icon class=\\"gux-expand-icon hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M11 6L8 9 5 6 4 7l4 4 4-4z\\"></path></svg></div></gux-icon></button><ul class=\\"gux-listbox\\" role=\\"listbox\\" tabindex=\\"-1\\" data-popper-placement=\\"bottom-start\\" style=\\"max-height: initial; width: 736px; position: absolute; left: 0px; top: 0px; margin: 0px; right: auto; bottom: auto; transform: translate(56px, 69px);\\"><li id=\\"gux-input-select-p94cs922xb-option-0\\" class=\\"gux-listbox-option\\" role=\\"option\\" aria-selected=\\"\\">Option 1</li><li id=\\"gux-input-select-p94cs922xb-option-1\\" class=\\"gux-listbox-option\\" role=\\"option\\">Option 2</li><li id=\\"gux-input-select-p94cs922xb-option-2\\" class=\\"gux-listbox-option\\" role=\\"option\\">Option 3</li></ul></div></gux-input-select></div><div class=\\"gux-error\\"></div></div>"`;
